src/EditorFeatures/Test/Diagnostics/DiagnosticAnalyzerServiceTests.cs

```diff
@@ -63,11 +63,10 @@ public async Task TestHasSuccessfullyLoadedBeingFalse()
         var document = GetDocumentFromIncompleteProject(workspace);
 
         var exportProvider = workspace.Services.SolutionServices.ExportProvider;
-        var service = Assert.IsType<DiagnosticAnalyzerService>(exportProvider.GetExportedValue<IDiagnosticAnalyzerService>());
-        var analyzer = service.CreateIncrementalAnalyzer(workspace);
+        var service = exportProvider.GetExportedValue<IDiagnosticAnalyzerService>();
         var globalOptions = exportProvider.GetExportedValue<IGlobalOptionService>();
 
-        var diagnostics = await analyzer.GetDiagnosticsForIdsAsync(
+        var diagnostics = await service.GetDiagnosticsForIdsAsync(
```

Comment:
removed bizarre testing of internal implementation details.  Tests should go through the service, not bypass it.

Reply:
in a followup, i will make DiagnosticIncrementalAnalyzer private and a nested type inside DIagnosticAnalyzerService.